### PR TITLE
feat(picker): configure home match prioritization

### LINKED ISSUE #105: feat: allow disabling native picker home prioritization
## Problem Statement

The native picker gives the actual home-directory session special treatment when the query is exactly `home`: it is included as a match and promoted ahead of ordinary workspace-name and path matches. This behavior is always enabled, so users who prefer normal name-first substring ordering cannot disable the home session's dedicated first-place priority through configuration.

## Solution

Add a native picker configuration setting named `prioritize_home`. It defaults to `true`, preserving current behavior. When set to `false`, the exact case-insensitive `home` query still recognizes the actual home-directory session as a searchable alias, but that session is classified with path-only matches instead of being promoted into a dedicated first bucket. Genuine workspace-name matches therefore remain ahead of it, while the configured workspace ordering remains stable within each match bucket.

Example native configuration:

```toml
[picker]
prioritize_home = false
```

## User Stories

1. As a native-picker user, I want to disable special home-session prioritization, so that `home` search results follow the same name-first ordering I expect from other queries.
2. As an existing user, I want home prioritization to remain enabled by default, so that upgrading does not unexpectedly change my picker ordering.
3. As a user with a workspace whose name contains `home`, I want that genuine name match to rank ahead of the actual home directory when prioritization is disabled, so that the workspace I named is easier to select.
4. As a user who disables home prioritization, I still want the actual home directory to match the exact `home` query, so that disabling its priority does not remove the useful home alias.
5. As a user, I want `home` matching and configuration behavior to remain case-insensitive in the same way as today, so that queries such as `HOME` behave consistently.
6. As a user with a configured workspace sort mode, I want that ordering preserved stably within name-match and path-only buckets, so that disabling home priority does not introduce unrelated reordering.
7. As a user with separator-aware matching enabled or disabled, I want that setting to keep its existing semantics, so that the new option affects only home prioritization.
8. As a user searching across Herdr workspaces, configured sessions, zoxide entries, and directories, I want the same per-session ranking rule applied consistently, so that source type does not change the meaning of the setting.
9. As a user entering any query other than exact `home`, I want filtering and ranking to remain unchanged, so that the option has a narrow and predictable effect.
10. As a user whose home path itself contains the substring `home`, I want the session treated as an ordinary path match when prioritization is disabled, so that it does not regain special priority accidentally.
11. As a user editing native configuration, I want the new key accepted by strict validation, so that I can validate the configuration before launching the picker.
12. As a user who explicitly configures `prioritize_home = false`, I want that false value preserved through configuration loading and merging, so that defaults cannot silently override my choice.
13. As a user reading the configuration documentation, I want the default and disabled behavior described with an example, so that I can choose the desired ordering without reading source code.
14. As an FZF user, I want the external picker behavior to remain unchanged, so that a native-picker-specific option does not alter the separate FZF search path.
15. As a maintainer, I want a focused regression at the shared native-picker filtering seam, so that future ranking changes cannot silently reintroduce unconditional home promotion.

## Implementation Decisions

- Add `prioritize_home` to the versioned native `[picker]` configuration table.
- Default the normalized runtime setting to `true` to preserve backward compatibility for omitted configuration.
- Preserve an explicitly configured `false` through decoding and configuration-layer merging; the source representation must distinguish omission from an explicit false where required by the existing merge model.
- Pass the normalized setting into the native picker model rather than branching in individual session sources or rendering code.
- Keep the existing exact, case-insensitive `home` detection and actual-home-path recognition.
- When prioritization is enabled, retain the current stable bucket order: actual home-directory session, real workspace-name matches, then path-only matches.
- When prioritization is disabled, omit the dedicated home bucket. Keep the actual home-directory session eligible through the home alias, but classify it as path-only; real workspace-name matches therefore precede it.
- Continue classifying each session from its own real name and path. Worktree metadata must not promote related sessions between match buckets.
- Preserve configured workspace ordering within every bucket.
- Keep query-change selection reset, match highlighting, Unicode handling, separator-aware matching, and all non-`home` filtering behavior unchanged.
- Keep the FZF picker and JSON output unchanged; this setting controls only native-picker ranking.
- Document the option, its default, and the distinction between retaining the home alias and disabling its priority.
- Do not add a new abstraction or ranking engine. Extend the existing shared filtering model with the minimum configuration input needed for this decision.

## Testing Decisions

- Use the shared native-picker filtering model as the single behavioral test seam because every native-picker source passes through it and existing ranking regressions already establish the home/name/path contract there.
- Add a focused test showing that the default behavior still promotes the actual home-directory session for an exact `home` query.
- Add a focused test showing that, with prioritization disabled, a genuine workspace-name match ranks ahead of the actual home-directory session while the home session remains present as a path-only match.
- Include stable ordering assertions for multiple name and path matches rather than asserting internal slice construction or helper names.
- Reuse the existing environment-isolation pattern for the home directory so the regression does not depend on the developer machine's path.
- Add narrow native-configuration coverage proving that omission resolves to `true` and an explicit `false` survives decoding into the runtime setting. This is schema/default coverage, not a second behavioral ranking seam.
- Keep existing filter regressions green for empty queries, both-field matches, unnamed path-label fallbacks, worktree relationships, separator-aware matching, selection reset, and non-home queries.
- Run the focused picker and configuration tests, followed by the repository aggregate check and whitespace validation.
- Tests should assert user-observable result inclusion and ordering, not the existence of a particular helper, bucket variable, or loop structure.

## Out of Scope

- Removing the `home` search alias entirely.
- Changing what counts as the actual home directory.
- Adding weighted relevance, fuzzy scoring, or configurable ranking weights.
- Changing workspace sort modes or their cycle order.
- Changing worktree-family presentation or grouping.
- Changing FZF search behavior or its separate home alias implementation.
- Changing JSON list output.
- Adding this native-picker-specific option to the legacy Sesh configuration schema.

## Further Notes

The option name describes the behavior being controlled rather than its implementation. A positive setting with a `true` default keeps generated documentation understandable while retaining compatibility. The intentionally narrow behavior is: keep `home` searchable, but let users opt out of always placing the actual home-directory session first.

### PR DESCRIPTION
**Pull request by fullerzz**
## Summary

- add native `[picker].prioritize_home` configuration with a backward-compatible `true` default
- keep the exact `home` alias while allowing real workspace-name matches to rank first when prioritization is disabled
- preserve stable bucket order and zero-value picker API behavior, with focused config, app, and picker regressions
- document the new setting and disabled behavior

Closes #105

## Validation

- `go test` focused RED/GREEN regressions for native config, app option wiring, picker filtering, and zero-value picker options
- `GOLANGCI_LINT_CACHE=/private/tmp/herdr-sesh-issue-105-final-lint-cache just check` (389 tests)
- `just build-docs`
- `git diff --check 65cbdbd..HEAD`

## Review

- task-scoped spec and quality review completed
- disabled actual-home name-match edge case fixed and re-reviewed
- whole-branch review completed; zero-value `picker.Options` compatibility fixed and re-reviewed

<a href="https://cubic.dev/pr/fullerzz/herdr-plugin-sesh/pull/106?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: internal/picker/model.go [hunk 1]
```diff
@@ -14,12 +14,14 @@ type Model struct {
 	Query          string
 	Selected       int
 	SeparatorAware bool
+	PrioritizeHome bool
```
[the remark this answers is not in the document]
Comment (fullerzz) [reply internal/picker/model.go]: Fixed in `5133707`. Replaced the positive Model flag with zero-value-safe `DisableHomePrioritization`, so exported `Model{}` preserves promotion and explicit disablement remains available. Added `TestZeroValueModelPrioritizesHome`; focused tests and `just check` (390 tests) pass.
